spec/features/item_seasons_spec.rb

```diff
@@ -3,14 +3,9 @@
 require 'rails_helper'
 
 RSpec.describe 'Item Seasons', type: :feature do
-  let(:produce_item) do
-    create(
-      :produce_item,
-      name: 'morango',
-      category: 'fruit',
-      item_seasons: [create(:item_season, produce_item: produce_item)]
-    )
-  end
+  let(:produce_item) { create(:produce_item) }
+
+  before { create(:item_season, produce_item: produce_item) }
 
```

Comment:
@tcione This doesn't seem to work. The item_season is returned(?) as nil. (Here I tried with create on line 11, but I also tried with the original build)

Question: We have the `produce_item_id` inside the `item_season`, how does the `item_seasons` array inside `produce_item` works? Looking at the schema I don't think we are able to access that information like this, but I'm not sure 😅 .

Also, while working on this, I realized we don't need to pass the name and category to `produce_item`

Reply:
> Also, while working on this, I realized we don't need to pass the name and category to produce_item, our factories are doing that for us, so there is no need to overwrite them here.

Quite happy that you picked that up <3

> The old version doesn't seem to work. The item_season is returned(?) as nil. (in that version, I tried with create on line 11, but I also tried with the original build)

I lost reference to what the old version was. Would you mind pasting it here?

> Question: We have the produce_item_id inside the item_season, how does the item_seasons array inside produce_item work? Looking at the schema I don't think we are able to access that information like this, but I'm not sure 😅 .

It's a bit of FactoryBot + Rails magic. If you pass an array of `ItemSeason` to `item_seasons`, it know to:

a. Persist the produce item
b. Attach this item id to seasons

But for that to work, you need to initiatize `ItemSeason` without persisting (so, using `build` on FactoryBot and just `new` when doing pure Rails)

If you use create, it will mess up the persistance order.

Reply:
Sorry, I think I wasn't clear, by the old version I meant your suggestion (since I changed it, now it is the old version 😅 ), this one:

```
  let(:produce_item) do
    create(
      :produce_item,
      name: 'morango',
      category: 'fruit',
      item_seasons: [create(:item_season, produce_item: produce_item)]
    )
 end
```

> If you use `create`, it will mess up the persistance order.

I used build at first, but it didn't work as well

> ... and just `new` when doing pure Rails)

What do you mean by this?

Reply:
About using arrays: we shouldn't be setting `produce_item`, FactoryBot/Rails should that for us, so either:
```rb
  let(:produce_item) do
    create(
      :produce_item,
      name: 'morango',
      category: 'fruit',
      item_seasons: [build(:item_season, produce_item: nil)]
    )
 end
```
or 
```rb
  let(:produce_item) do
    create(
      :produce_item,
      name: 'morango',
      category: 'fruit',
      item_seasons: [build(:item_season)]
    )
 end
```
(not sure which would work).

---

> ... and just new when doing pure Rails)
>
> What do you mean by this?

`ItemSeason.new(...)`

Reply:
Hum, I think I didn't express myself properly, I understand its `ItemSeason.new(...)`, I don't understand how everything here works together